composition-js/src/__tests__/compose.test.ts

```diff
@@ -4008,4 +4009,287 @@ describe('composition', () => {
       assertCompositionSuccess(result);
     });
   });
+
+  describe('@authenticated', () => {
+    it('comprehensive locations', () => {
+      const onObject = {
+        typeDefs: gql`
+          type Query {
+            object: AuthenticatedObject!
+          }
+
+          type AuthenticatedObject @authenticated {
+            field: Int!
+          }
+        `,
+        name: 'on-object',
+      };
+
+      const onInterface = {
+        typeDefs: gql`
+          type Query {
+            interface: AuthenticatedInterface!
+          }
+
+          interface AuthenticatedInterface @authenticated {
+            field: Int!
+          }
+        `,
+        name: 'on-interface',
+      };
+
+      const onScalar = {
+        typeDefs: gql`
+          scalar AuthenticatedScalar @authenticated
+        `,
+        name: 'on-scalar',
+      };
+
+      const onEnum = {
+        typeDefs: gql`
+          enum AuthenticatedEnum @authenticated {
+            A
+            B
+          }
+        `,
+        name: 'on-enum',
+      };
+
+      const onRootField = {
+        typeDefs: gql`
+          type Query {
+            authenticatedRootField: Int! @authenticated
+          }
+        `,
+        name: 'on-root-field',
+      };
+
+      const onObjectField = {
+        typeDefs: gql`
+          type Query {
+            objectWithField: ObjectWithAuthenticatedField!
+          }
+
+          type ObjectWithAuthenticatedField {
+            field: Int! @authenticated
+          }
+        `,
+        name: 'on-object-field',
+      };
+
+      const onEntityField = {
+        typeDefs: gql`
+          type Query {
+            entityWithField: EntityWithAuthenticatedField!
+          }
+
+          type EntityWithAuthenticatedField @key(fields: "id") {
+            id: ID!
+            field: Int! @authenticated
+          }
+        `,
+        name: 'on-entity-field',
+      };
+
+      const result = composeAsFed2Subgraphs([
+        onObject,
+        onInterface, onScalar, onEnum, onRootField, onObjectField, onEntityField,
+      ]);
+      assertCompositionSuccess(result);
+      expect(result.supergraphSdl.includes(dedent`
```

Comment:
Fwiw, I usually advocate for avoiding to test full supergraph sdl like here, unless the test is really about that full sdl. And this for 2 reasons:
1. it's not super targeted, so it takes a bit more time for a future reader to find what this test really test versus what is other noises.
2. it gets a bit annoying when we do some minor change to the supergraph format (saying bumping the join spec to add some new argument to an existing `@join__` directive) and we have to modify a million unrelated test just because they match the full sdl.

Tbc, this is more of a nitpick and not insisting at all, just mentioning it as it occurred to me.

Fwiw, what I think matter to this test is that the subgraphs compose, and that the element with `@authenticated` in the subgraph are properly marked in the supergraph. So the assertions I'd probably use here would be something like:
```typescript
expect(result.supergraph.elementByCoordinate('AuthenticatedObject')!).hasAppliedDirective('authenticated')).toBeTruthy();
expect(result.supergraph.elementByCoordinate('EntityWithAuthenticateField.field')).hasAppliedDirective('authenticated')).toBeTruthy();
...
```

Reply:
Good nitpick, much prefer this